### PR TITLE
Usability fixes for new room header

### PR DESCRIPTION
**Pull request by germain-gg**
Fixes a lot of UI feedback shared as part of https://www.figma.com/file/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?type=design&node-id=1609-8863&mode=design&t=hw2eZG1msBjOombN-11
Fixes https://github.com/vector-im/element-web/issues/26346

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: res/css/views/rooms/_RoomHeader.pcss [hunk 1]
```diff
@@ -19,30 +19,41 @@ limitations under the License.
...
+}
+
 .mx_RoomHeader_heading {
     display: flex;
     gap: var(--cpd-space-1x);
     align-items: center;
+
+    /* Spacing of 64px between the topic and the buttons, but we have
+       to account for the gap of the flex items */
+    padding-right: var(--cpd-space-13x);
```
Comment (t3chguy) [inline res/css/views/rooms/_RoomHeader.pcss:51]: Something is a bit strange with a large gap growing at low widths

I don't see this gap in the Figma comps

image

Comment (t3chguy) [reply res/css/views/rooms/_RoomHeader.pcss:51]: The topic fills the space fine

image

File: res/css/views/rooms/_RoomHeader.pcss [hunk 2]
```diff
@@ -29,6 +29,9 @@ limitations under the License.
 
 .mx_RoomHeader_info {
     flex: 1;
+    /* Spacing of 64px between the topic and the buttons, but we have
+       to account for the gap of the flex items */
+    padding-right: var(--cpd-space-13x);
```
Comment (t3chguy) [inline res/css/views/rooms/_RoomHeader.pcss:34]: image

Based on the comment it looks like the wrong variable is used

**Review by t3chguy — APPROVED** [review]
The gap being so big in narrow windows looks like waste a lot of space but at least its consistent, lgtm other than comment clarity

File: res/css/views/rooms/_RoomHeader.pcss (hunk 2 above)
Comment (germain-gg) [reply res/css/views/rooms/_RoomHeader.pcss:34]: The gap on the flex parent is `--cpd-space-3x` which is `12px`.  It all adds up to 64px. Might cahnge the owrding of the comment

**Pull request merged**